Document requirements to use network areas (#12290)
hashicorp/consul: website/content/docs/connect/gateways/mesh-gateway/wan-federation-via-mesh-gateways.mdx

# WAN Federation via Mesh Gateways

> -> **1.8.0+:** This feature is available in Consul versions 1.8.0 and higher

Author: This was moved into the requirements section. Is there a reason it needs to stay here?

Reviewer: None that I can see -- I think that's a fine change.

> Cross-DC control plane communication, also known as WAN federation,
> allows for Consul servers in different datacenters to be federated exclusively through mesh gateways.

Reviewer: ```suggestion
This topic describes how to enable cross-datacenter communication, also referred to as WAN federation, in the  control plane. WAN federation allows you to designate a primary datacenter and one or more secondary datacenters. In this configuration, the Consul servers in different datacenters are connected exclusively through mesh gateways.

## Background
```
I wanted to do a few things here: 

1. Use language that specifically calls out what the reader will get out of the topic (i.e. "This topic describes . . .")
2. Use the spelled-out "datacenter" term (as opposed to the abbreviation "DC") so that we don't confuse readers.
3. Try to include a brief definition of what we mean by "federation".    
4. Create a subheading for the following information. I'm not sure if "Background" is the best term. I would consider moving this information into the "Requirements" section with its own subheading.

To me, the graphics are a bit overwhelming, especially upfront. Would it make more sense to move them to the "Architecture" section?

Author: I tried to make a more concise version of the suggested first three sentences (that open the topic). Let me know what you think.

```
This topic describes how to configure Consul servers in federated datacenters to
communicate exclusively through mesh gateways.
```

Author: > To me, the graphics are a bit overwhelming, especially upfront. Would it make more sense to move them to the "Architecture" section?

I'll play with options for this, but I might want to address those in a separate PR if we don't come to a solution we like quickly. I don't want that to block this change to list requirements (which I want to use as a template for similar "requirements" content on other pages).

Author: And I haven't thought of a better subheading title than "Background", so I've left it as that for now.

Author: I think the images belong in the Background section because they directly pair with that text (which talks about the challenges of traditional WAN federation and how mesh gateways help).

One thing we could do is move these images to the bottom of the section and have a "before/after" style tabset. I'm not sure I like this more than the current approach, but wanted to present it for feedback:

![image](https://user-images.githubusercontent.com/85913323/155228216-68d5796e-577d-4ad0-9de1-bf5caf99d521.png)

# WAN Federation via Mesh Gateways

~> This topic requires familiarity with [mesh gateways](/docs/connect/gateways/mesh-gateway/service-to-service-traffic-datacenters).

This topic describes how to configure Consul servers in federated datacenters to
communicate exclusively through mesh gateways.

## Background

When setting up a

> [multi-datacenter](https://learn.hashicorp.com/tutorials/consul/federarion-gossip-wan)

Author: > I suppose this won't be an issue soon enough.

Why will this become less of an issue soon?

[![WAN federation with mesh gateways](/img/wan-federation-connectivity-mesh-gateways.png)](/img/wan-federation-connectivity-mesh-gateways.png)

## Requirements

- Consul 1.8.0+

> - Mesh federation between Consul datacenters: all datacenters must be connected to all other datacenters

Reviewer: ```suggestion
- Mesh federation between Consul datacenters: all datacenters must be connected to all other datacenters
```
I'm a little confused by this bullet--isn't this topic about how to enact this requirement?

Author: I think I know what the confusion is. It's something I was worried about when writing this sentence, but didn't have an alternative ready at hand.

"Mesh" is being used for multiple things here, and that makes it confusing. I should probably describe this differently.

"Mesh federation" refers to a topology in which every datacenter is connected to every other datacenter. (I'm not sure this is a commonly understood term, but we do use it elsewhere in our docs.) You can accomplish "mesh federation" with or without "mesh gateways"... Which is confusing.

I think I need to remove the term "Mesh federation" from this. Maybe I can just simplify to:
```suggestion
- All Consul datacenters must be connected to all other datacenters
```

Thoughts?

Reviewer: I think I understand now--the overloaded term "mesh" is probably a big reason why users have trouble understanding our docs. The change makes this a little clearer.

Reviewer 2: > "Mesh federation" refers to a topology in which every datacenter is connected to every other datacenter. (I'm not sure this is a commonly understood term, but we do use it elsewhere in our docs.)

I'm unable to find any references to "mesh federation" on consul.io or Learn. Would you mind sharing where that terminology is used?

There are sections of the documentation, such as the [Advanced Federation](https://www.consul.io/docs/enterprise/federation) page, that state that Consul's traditional federation model requires that… "every datacenter be in a fully connected mesh." This referring to a [fully connected network](https://en.wikipedia.org/wiki/Network_topology#Fully_connected_network) which is a type of mesh network topology.

> the overloaded term "mesh" is probably a big reason why users have trouble understanding our docs.

I wouldn't consider the term to be overloaded, since its technically correct. I think that in order to reduce confusion between mesh networking topologies and service mesh, it might be better if our docs replace "fully connected mesh" with "fully connected network."

Author: I think in my memory "fully connected mesh" morphed into "mesh federation" for some reason. You're right.

I agree that "fully connected network" would be less confusing than "fully connected mesh" in this context in particular, and probably in other contexts in our docs as well.

Author: Changed to:

```
- All Consul datacenters must be connected to all other datacenters
   - Other topologies, such as "hub-and-spoke" enabled by Consul Enterprise network areas,
     are not supported
```

>    - Other topologies, such as [hub-and-spoke enabled by Consul Enterprise network areas](https://learn.hashicorp.com/tutorials/consul/federation-network-areas),

Reviewer: ```suggestion
   - Other topologies, such as "hub-and-spoke" enabled by Consul Enterprise network areas, are not supported. Refer to the [Consul federation tutorial](https://learn.hashicorp.com/tutorials/consul/federation-network-areas) for a discussion about hub-and-spoke topologies.
```
I think it would be a good practice to start explicitly calling out when we are sending readers to content outside of Consul.io. It's just a small thing I think we can do to help orient readers when they are about to go to a different site.

Author: I would imagine it's generally true that it would be disorienting to be redirected to a different site. Do we have a sense of whether our users see our docs and Learn sites as "not one thing" (so the redirect is disorienting)? Or that they are seen as "one thing" that happens to be on two different domains with slightly different look and feel?

(This probably relates to longer-term thinking about the division of responsibilities between those two sites.)

Reviewer: From what I understand, users generally view them as one thing, which is probably adds to the confusion when we link directly across the sites without some sort of heads up.

Author: In a different comment (on `federation.mdx`), you recommended:
> I think we should remove the link to information we're telling users not to implement.

Does that mean we should also not link to a hub-and-spoke guide here?

>      cannot be used

Reviewer: ```suggestion

```

Author: I'm confused, why remove `cannot be used`?

Reviewer: My previous suggestion would have removed it: https://github.com/hashicorp/consul/pull/12290#discussion_r801932914

[![WAN federation with mesh gateways](/img/wan-federation-connectivity-mesh-gateways.png)](/img/wan-federation-connectivity-mesh-gateways.png)

## Requirements

> - Consul 1.8.0+
> - All Consul datacenters must be connected to all other datacenters
>    - Other topologies, such as "hub-and-spoke" enabled by Consul Enterprise network areas,
>      are not supported

Reviewer: ```suggestion
- Consul 1.8.0+
- All Consul datacenters must be connected to all other datacenters. Other topologies, such  as _hub-and-spoke_ enabled by Consul Enterprise network areas, are not supported.
```
We had a contentious discussion amongst the TWs and one of the hot-button issues was when to use quotation marks and when to use italics. We will be updating the guidance to something like:

* Use quotation marks when required as a part of the code syntax and when referring to titles of books, articles, and other works. 
* Do not use quotation marks to introduce terms or to add emphasis. Use italics instead. 

Sorry for the back and forth.

I think the second requirement might work better as a single bullet because it's a qualification/clarification of the initial statement, rather than a sub requirement.

> - Consul 1.8.0+
> - All Consul datacenters must be connected to all other datacenters
>    - Other topologies, such as "hub-and-spoke" enabled by Consul Enterprise network areas,
>      are not supported

Reviewer: ```suggestion
- Consul 1.8.0+
- All Consul datacenters must be connected to all other datacenters. Other topologies, such  as _hub-and-spoke_  enabled by Consul Enterprise network areas, are not supported.
```
We had a contentious discussion amongst the TWs and one of the hot-button issues was when to use quotation marks and when to use italics. We will be updating the guidance to something like:

* Use quotation marks when required as a part of the code syntax and when referring to titles of books, articles, and other works. 
* Do not use quotation marks to introduce terms or to add emphasis. Use italics instead. 

Sorry for the back and forth.

I think the second requirement might work better as a single bullet because it's a qualification/clarification of the initial statement, rather than a sub requirement.
